x-pack/plugins/monitoring/dev_docs/reference/data_collection_modes.md

```diff
@@ -0,0 +1,199 @@
+
+```mermaid
+graph TD
+subgraph Production Deployment
+  ProdElasticsearch[Elasticsearch]
+end
+
+subgraph Monitoring Deployment
+  MonElasticsearch[Elasticsearch]
+end
+
+subgraph Kibana
+  Collectors-.->|poll|Metrics[OS, Cgroups, etc]
+  Collectors-.->|update|OpsMetricsObservable
+  BulkUploader-.->|observe|OpsMetricsObservable
+  
+  click Collectors "https://github.com/elastic/kibana/tree/main/src/core/server/metrics/collectors"
+  click OpsMetricsObservable "https://github.com/elastic/kibana/blob/92a8636f0ff63ab072527574e96e6616327b2ea4/src/core/server/metrics/metrics_service.ts#L32"
+  click BulkUploader "https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/server/kibana_monitoring/bulk_uploader.ts"
+end 
+
+BulkUploader-->|/_monitoring/bulk|ProdElasticsearch
+ProdElasticsearch-->|/_bulk|MonElasticsearch
+```
+
+The production cluster receives these metrics on the `/_monitoring/bulk` endpoint in elasticsearch, then forwards them to the [monitoring deployment](../reference/terminology.md#monitoring-deployment) along with its own metrics. Or writes them directly to disk if only using a single deployment.
+
+Alternatively some stack components (logstash & beats) can be configured to send data directly to the [monitoring deployment](../reference/terminology.md#monitoring-deployment) either using a user-provided cluster UUID, or sending the data with no UUID to be associated with the [standalone cluster](../reference/terminology.md#standalone-cluster).
+
+Here's an example from Logstash, which is capable of sending data directly to a monitoring deployment. It will associate the data with the UUID of any elasticsearch clusters found in the pipeline outputs.
+
+```mermaid
+graph LR
+subgraph Logstash
+  Pipeline-.->|output uuid|MonitoringExtension
+  MonitoringExtension
+  click MonitoringExtension "https://github.com/elastic/logstash/blob/main/x-pack/lib/monitoring/monitoring.rb"
+end
+
+subgraph Monitoring Deployment
+  MonElasticsearch[Elasticsearch]
+end
+MonitoringExtension-->|_bulk|MonElasticsearch
+```
+
+### Standalone Metricbeat collection
+
+When using standalone metricbeat collection, each component in your Elastic stack exposes an endpoint for metricbeat to collect metric data.
+
+Each component has a corresponding metricbeat module that will read the endpoint and publish data directly to the [monitoring deployment](../reference/terminology.md#monitoring-deployment).
+
+- [Elasticsearch](https://github.com/elastic/beats/tree/main/metricbeat/module/elasticsearch)
+- [Kibana](https://github.com/elastic/beats/tree/main/metricbeat/module/kibana)
+- [Logstash](https://github.com/elastic/beats/tree/main/metricbeat/module/logstash)
+- [Beats](https://github.com/elastic/beats/tree/main/metricbeat/module/beat)
+- [Enterprise Search](https://github.com/elastic/beats/tree/main/x-pack/metricbeat/module/enterprisesearch)
```

Comment:
Where does APM/Fleet server fit in here?

Reply:
Ah, I should add a note to the blurb a the top of the doc. apm-server is there, fleet isn't. Both are "beats" so I group them together in places like these.